### PR TITLE
VSTS-171 - Write SONAR_SCANNER_OPTS env variable if standalone scanner is selected in the Prepare Analysis Task configuration.

### PR DESCRIPTION
**Pull request by mickael-caro-sonarsource**
This issue was because the call to sonar-scanner cli is done through a wrapper (.bat or not), using SONAR_SCANNER_OPTS environment variable to fill arguments to pass to that CLI. This env variable was never written in the extension, so now it's done.

### CODE REVIEW

**Comment by mickael-caro-sonarsource** [comment]
Well, that has been done !

**Comment by maikelgithub** [comment]
Hi,

since the update to 4.7.0 was released this morning, we can't run SonarQube scans anymore.
This is our configuration: Azure DevOps Pipeline with self hosted agents.

Until this morning, all our scans we sent to SonarQube without any issues.

The problem:
`
##[error]Cannot read property 'concat' of undefined 
`
- This is happening in the Prepare analysis on SonarQube task

Further config details:
- standalone scanner
- manually provide configuration

- additional properties:
```
sonar.log.level=ERROR
sonar.verbose=false
sonar.issue.ignore.multicriteria=e1,e2
sonar.issue.ignore.multicriteria.e1.ruleKey=javascript:S3317
sonar.issue.ignore.multicriteria.e1.resourceKey=**/*.js
sonar.issue.ignore.multicriteria.e2.ruleKey=javascript:TrailingWhitespace
sonar.issue.ignore.multicriteria.e2.resourceKey=**/*.js 
```

[We found the related line causing the issue, according to our debugging](https://github.com/SonarSource/sonar-scanner-vsts/blob/ca7ad3910ef2ebc8d09cecd4f404b343321cca24/common/ts/prepare-task.ts#L57).

Is it possible to undo this merge, or how do we fix the issue?

**Comment by mickael-caro-sonarsource** [comment]
Hi, 

A fix will be deployed this afternoon. In the mean time, as a workaround, you can set by and, before the Prepare analysis task, the SONAR_SCANNER_OPTS env variable by hand.

Sorry for the inconvenience.

**Comment by clement-fleury-assoconnect** [comment]
Dude, this has been really painful for a lot of people https://developercommunity.visualstudio.com/content/problem/692109/pipeline-task-prepare-sonarcloud-analysis-configur.html

**Comment by clement-fleury-assoconnect** [comment]
@mickael-caro-sonarsource let us know when this is fixed and deployed

**Comment by mickael-caro-sonarsource** [comment]
Hi @clement-fleury-assoconnect , it has just been deployed : version 1.8.1 for SonarCloud, 4.7.1 for SonarQube.

**Comment by clement-fleury-assoconnect** [comment]
Thanks, @michal-duda-sonarsource , this issue is fixed for me ! 👍

**Comment by michal-duda-sonarsource** [comment]
Glad it's working for you @clement-fleury-assoconnect but all the credit should go to @mickael-caro-sonarsource , I had nothing to do with this fix 😃

File: common/ts/prepare-task.ts [hunk 1]
```diff
@@ -46,6 +46,10 @@ export default async function prepareTask(endpoint: Endpoint, rootPath: string)
     })
   );
 
+  if (scannerMode === ScannerMode.CLI) {
+    tl.setVariable('SONAR_SCANNER_OPTS', scanner.toCliProps());
```
Comment (henryju) [inline common/ts/prepare-task.ts]: I'm really surprised by this change. SONAR_SCANNER_OPTS is intended to pass JVM arguments, not scanner properties. What if users already try to set SONAR_SCANNER_OPTS to increase memory (like -Xmx)?

**Pull request merged**